File: tests/fixtures/sample.ts
interface Point {
  x: number;
  y: number;
}

type Pair = [number, number];

enum Direction {
  Up,
  Down,
  Left,
  Right,
}

class Vector2D {
  constructor(public x: number, public y: number) {}

  add(other: Point): Vector2D {
    return new Vector2D(this.x + other.x, this.y + other.y);
  }

  magnitude(): number {
    return Math.sqrt(this.x * this.x + this.y * this.y);
  }
}

function distance(a: Point, b: Point): number {
  const dx = a.x - b.x;
  const dy = a.y - b.y;
  return Math.sqrt(dx * dx + dy * dy);
}

const normalize = (v: Point): Point => {
  const len = Math.sqrt(v.x * v.x + v.y * v.y);
  return { x: v.x / len, y: v.y / len };
};

File: tests/fixtures/sample.zig
const std = @import("std");

const Point = struct {
    x: f64,
    y: f64,
};

const Direction = enum {
    up,
    down,
    left,
    right,
};

const Result = union(enum) {
    ok: i32,
    err: []const u8,
};

const OpaqueType = opaque {};

pub fn add(a: i32, b: i32) i32 {
    return a + b;
}

fn greet(name: []const u8) []const u8 {
    return "Hello, " ++ name ++ "!";
}

pub fn magnitude(p: Point) f64 {
    return @sqrt(p.x * p.x + p.y * p.y);
}
